### PR TITLE
Move cockpit draw code to cockpit_gfx

### PR DESCRIPTION
**Pull request by royfalk**
Also try to fix IDE complaints about unknown size_t and FLT_MAX by adding relevant headers. Doesn't seem to solve it at my end but leaving anyway, as it's good practice.

This replaces the much more ambitious but ultimately unsuccessful #655. I've copied some of the code from it and packaged it in this PR.

Please answer the following:

Code Changes:
- [Yes] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation

Issues:
- None

Purpose:
- What is this pull request trying to do? Just general code refactoring
- What release is this for? Any
- Is there a project or milestone we should apply this to? No

### CODE REVIEW

**Comment by stephengtuggy** [comment]
The Windows build failed with the following two errors:

```
D:\a\Vega-Strike-Engine-Source\Vega-Strike-Engine-Source\engine\src\gfx\cockpit_gfx_utils.cpp(209,27): error C2065: 'M_PI': undeclared identifier [D:\a\Vega-Strike-Engine-Source\Vega-Strike-Engine-Source\build\vegastrike-engine.vcxproj]
  D:\a\Vega-Strike-Engine-Source\Vega-Strike-Engine-Source\engine\src\gfx\cockpit_gfx_utils.cpp(209,17): error C2737: 'theta': const object must be initialized [D:\a\Vega-Strike-Engine-Source\Vega-Strike-Engine-Source\build\vegastrike-engine.vcxproj]
```

We'll need to fix those before merging this.

**Comment by stephengtuggy** [comment]
That latest change fixed the Windows build! Sweet.

File: engine/src/gfx/cockpit.cpp [hunk 1]
```diff
@@ -2618,9 +1664,9 @@ void GameCockpit::Draw() {
                 headtrans.back().p.j = shake_mag * shakin * cos(1.3731 * theta) * cockpitradial
                         / 100;  //AccessCamera()->GetPosition().j+shakin*cos(theta);
                 headtrans.back().p.k = 0;                 //AccessCamera()->GetPosition().k;
-                headtrans.back().p.i += warp_shake_mag * cos(wtheta) * sqr(warp_strength) * cockpitradial
+                headtrans.back().p.i += warp_shake_mag * cos(wtheta) * sqrtf(warp_strength) * cockpitradial
```
Comment (stephengtuggy) [inline engine/src/gfx/cockpit.cpp:1667]: Why `sqrtf`? Why not just `sqrt`?

File: engine/src/gfx/cockpit.cpp [hunk 2]
```diff
@@ -2618,9 +1664,9 @@ void GameCockpit::Draw() {
...
                         / 100;           //AccessCamera()->GetPosition().i+shakin*cos(theta);
-                headtrans.back().p.j += warp_shake_mag * cos(1.165864 * wtheta) * sqr(warp_strength) * cockpitradial
+                headtrans.back().p.j += warp_shake_mag * cos(1.165864 * wtheta) * sqrtf(warp_strength) * cockpitradial
```
Comment (stephengtuggy) [inline engine/src/gfx/cockpit.cpp:1669]: Same on this line

**Review by stephengtuggy — COMMENTED** [review]
I just had a question about a couple of lines. I'm not sure if I'll block for this, but it would be good to at least discuss it briefly

File: engine/src/gfx/cockpit.cpp (hunk 1 above)
Comment (royfalk) [reply engine/src/gfx/cockpit.cpp:1667]: It was originally a macro here in the previous MR and I replaced it with a standard function. I usually use std::sqrt in such places. For some reason, the IDE didn't like either that or sqrt and I didn't spend much time on it. 
It's possible the reason for it was that warp_strength is a float. Most variables in VS are floats for reasons of performance. I'm not sure if this is still a valid reason in today's architectures, but converting them en masse is asking for trouble.

File: engine/src/gfx/cockpit.cpp (hunk 2 above)
Comment (royfalk) [reply engine/src/gfx/cockpit.cpp:1669]: Same as above.

File: engine/src/gfx/cockpit.cpp (hunk 1 above)
Comment (stephengtuggy) [reply engine/src/gfx/cockpit.cpp:1667]: OK, we can leave these as is then.

**Review by stephengtuggy — APPROVED** [review]

**Pull request merged**